import random
import socket

from testing import protoTest_pb2


def genNonProto():
    temp = []
    for x in range(0,256):
        temp.append(random.randrange(0,256))
    return temp



def genProtoPacket():
    message = protoTest_pb2.telemetry()
    message.podState = 1
    message.sensor0Temp = random.randint(0, 256)
    message.sensor1Temp = random.randint(0, 256)
    message.sensor2Temp = random.randint(0, 256)
    message.sensor3Temp = random.randint(0, 256)
    message.sensor4Temp = random.randint(0, 256)
    message.sensor5Temp = random.randint(0, 256)
    message.sensor6Temp = random.randint(0, 256)
    message.sensor7Temp = random.randint(0, 256)
    message.sensor8Temp = random.randint(0, 256)
    message.sensor9Temp = random.randint(0, 256)
    message.sensor10Temp = random.randint(0, 256)
    message.sensor11Temp = random.randint(0, 256)
    message.sensor12Temp = random.randint(0, 256)
    message.sensor13Temp = random.randint(0, 256)
    message.sensor14Temp = random.randint(0, 256)
    message.sensor15Temp = random.randint(0, 256)
    message.sensor16Temp = random.randint(0, 256)
    message.sensor17Temp = random.randint(0, 256)
    message.sensor18Temp = random.randint(0, 256)
    message.sensor19Temp = random.randint(0, 256)
    message.sensor20Temp = random.randint(0, 256)
    message.sensor21Temp = random.randint(0, 256)
    message.sensor22Temp = random.randint(0, 256)
    message.sensor23Temp = random.randint(0, 256)
    message.sensor24Temp = random.randint(0, 256)
    message.sensor25Temp = random.randint(0, 256)
    message.sensor26Temp = random.randint(0, 256)
    message.sensor27Temp = random.randint(0, 256)
    message.sensor28Temp = random.randint(0, 256)
    message.sensor29Temp = random.randint(0, 256)
    message.sensor30Temp = random.randint(0, 256)
    message.sensor31Temp = random.randint(0, 256)
    message.sensor32Temp = random.randint(0, 256)
    message.sensor33Temp = random.randint(0, 256)
    message.sensor34Temp = random.randint(0, 256)
    message.sensor35Temp = random.randint(0, 256)
    message.sensor36Temp = random.randint(0, 256)
    message.sensor37Temp = random.randint(0, 256)
    message.sensor38Temp = random.randint(0, 256)
    message.sensor39Temp = random.randint(0, 256)
    message.sensor40Temp = random.randint(0, 256)
    message.sensor41Temp = random.randint(0, 256)
    message.sensor42Temp = random.randint(0, 256)
    message.sensor43Temp = random.randint(0, 256)
    message.sensor44Temp = random.randint(0, 256)
    message.sensor45Temp = random.randint(0, 256)
    message.sensor46Temp = random.randint(0, 256)
    message.sensor47Temp = random.randint(0, 256)
    message.sensor48Temp = random.randint(0, 256)
    message.sensor49Temp = random.randint(0, 256)
    message.sensor50Temp = random.randint(0, 256)
    message.sensor51Temp = random.randint(0, 256)
    message.sensor52Temp = random.randint(0, 256)
    message.sensor53Temp = random.randint(0, 256)
    message.sensor54Temp = random.randint(0, 256)
    message.sensor55Temp = random.randint(0, 256)
    message.sensor56Temp = random.randint(0, 256)
    message.sensor57Temp = random.randint(0, 256)
    message.sensor58Temp = random.randint(0, 256)
    message.sensor59Temp = random.randint(0, 256)
    message.sensor60Temp = random.randint(0, 256)
    message.sensor61Temp = random.randint(0, 256)
    message.sensor62Temp = random.randint(0, 256)
    message.sensor63Temp = random.randint(0, 256)
    message.sensor64Temp = random.randint(0, 256)
    message.sensor65Temp = random.randint(0, 256)
    message.sensor66Temp = random.randint(0, 256)
    message.sensor67Temp = random.randint(0, 256)
    message.sensor68Temp = random.randint(0, 256)
    message.sensor69Temp = random.randint(0, 256)
    message.sensor70Temp = random.randint(0, 256)
    message.sensor71Temp = random.randint(0, 256)
    message.sensor72Temp = random.randint(0, 256)
    message.sensor73Temp = random.randint(0, 256)
    message.sensor74Temp = random.randint(0, 256)
    message.sensor75Temp = random.randint(0, 256)
    message.sensor76Temp = random.randint(0, 256)
    message.sensor77Temp = random.randint(0, 256)
    message.sensor78Temp = random.randint(0, 256)
    message.sensor79Temp = random.randint(0, 256)
    message.sensor80Temp = random.randint(0, 256)
    message.sensor81Temp = random.randint(0, 256)
    message.sensor82Temp = random.randint(0, 256)
    message.sensor83Temp = random.randint(0, 256)
    message.sensor84Temp = random.randint(0, 256)
    message.sensor85Temp = random.randint(0, 256)
    message.sensor86Temp = random.randint(0, 256)
    message.sensor87Temp = random.randint(0, 256)
    message.sensor88Temp = random.randint(0, 256)
    message.sensor89Temp = random.randint(0, 256)
    message.sensor90Temp = random.randint(0, 256)
    message.sensor91Temp = random.randint(0, 256)
    message.sensor92Temp = random.randint(0, 256)
    message.sensor93Temp = random.randint(0, 256)
    message.sensor94Temp = random.randint(0, 256)
    message.sensor95Temp = random.randint(0, 256)
    message.sensor96Temp = random.randint(0, 256)
    message.sensor97Temp = random.randint(0, 256)
    message.sensor98Temp = random.randint(0, 256)
    message.sensor99Temp = random.randint(0, 256)
    message.sensor100Temp = random.randint(0, 256)
    message.sensor101Temp = random.randint(0, 256)
    message.sensor102Temp = random.randint(0, 256)
    message.sensor103Temp = random.randint(0, 256)
    message.sensor104Temp = random.randint(0, 256)
    message.sensor105Temp = random.randint(0, 256)
    message.sensor106Temp = random.randint(0, 256)
    message.sensor107Temp = random.randint(0, 256)
    message.sensor108Temp = random.randint(0, 256)
    message.sensor109Temp = random.randint(0, 256)
    message.sensor110Temp = random.randint(0, 256)
    message.sensor111Temp = random.randint(0, 256)
    message.sensor112Temp = random.randint(0, 256)
    message.sensor113Temp = random.randint(0, 256)
    message.sensor114Temp = random.randint(0, 256)
    message.sensor115Temp = random.randint(0, 256)
    message.sensor116Temp = random.randint(0, 256)
    message.sensor117Temp = random.randint(0, 256)
    message.sensor118Temp = random.randint(0, 256)
    message.sensor119Temp = random.randint(0, 256)
    message.sensor120Temp = random.randint(0, 256)
    message.sensor121Temp = random.randint(0, 256)
    message.sensor122Temp = random.randint(0, 256)
    message.sensor123Temp = random.randint(0, 256)
    message.sensor124Temp = random.randint(0, 256)
    message.sensor125Temp = random.randint(0, 256)
    message.sensor126Temp = random.randint(0, 256)
    message.sensor127Temp = random.randint(0, 256)
    message.sensor128Temp = random.randint(0, 256)
    message.sensor129Temp = random.randint(0, 256)
    message.sensor130Temp = random.randint(0, 256)
    message.sensor131Temp = random.randint(0, 256)
    message.sensor132Temp = random.randint(0, 256)
    message.sensor133Temp = random.randint(0, 256)
    message.sensor134Temp = random.randint(0, 256)
    message.sensor135Temp = random.randint(0, 256)
    message.sensor136Temp = random.randint(0, 256)
    message.sensor137Temp = random.randint(0, 256)
    message.sensor138Temp = random.randint(0, 256)
    message.sensor139Temp = random.randint(0, 256)
    message.sensor140Temp = random.randint(0, 256)
    message.sensor141Temp = random.randint(0, 256)
    message.sensor142Temp = random.randint(0, 256)
    message.sensor143Temp = random.randint(0, 256)
    message.sensor144Temp = random.randint(0, 256)
    message.sensor145Temp = random.randint(0, 256)
    message.sensor146Temp = random.randint(0, 256)
    message.sensor147Temp = random.randint(0, 256)
    message.sensor148Temp = random.randint(0, 256)
    message.sensor149Temp = random.randint(0, 256)
    message.sensor150Temp = random.randint(0, 256)
    message.sensor151Temp = random.randint(0, 256)
    message.sensor152Temp = random.randint(0, 256)
    message.sensor153Temp = random.randint(0, 256)
    message.sensor154Temp = random.randint(0, 256)
    message.sensor155Temp = random.randint(0, 256)
    message.sensor156Temp = random.randint(0, 256)
    message.sensor157Temp = random.randint(0, 256)
    message.sensor158Temp = random.randint(0, 256)
    message.sensor159Temp = random.randint(0, 256)
    message.sensor160Temp = random.randint(0, 256)
    message.sensor161Temp = random.randint(0, 256)
    message.sensor162Temp = random.randint(0, 256)
    message.sensor163Temp = random.randint(0, 256)
    message.sensor164Temp = random.randint(0, 256)
    message.sensor165Temp = random.randint(0, 256)
    message.sensor166Temp = random.randint(0, 256)
    message.sensor167Temp = random.randint(0, 256)
    message.sensor168Temp = random.randint(0, 256)
    message.sensor169Temp = random.randint(0, 256)
    message.sensor170Temp = random.randint(0, 256)
    message.sensor171Temp = random.randint(0, 256)
    message.sensor172Temp = random.randint(0, 256)
    message.sensor173Temp = random.randint(0, 256)
    message.sensor174Temp = random.randint(0, 256)
    message.sensor175Temp = random.randint(0, 256)
    message.sensor176Temp = random.randint(0, 256)
    message.sensor177Temp = random.randint(0, 256)
    message.sensor178Temp = random.randint(0, 256)
    message.sensor179Temp = random.randint(0, 256)
    message.sensor180Temp = random.randint(0, 256)
    message.sensor181Temp = random.randint(0, 256)
    message.sensor182Temp = random.randint(0, 256)
    message.sensor183Temp = random.randint(0, 256)
    message.sensor184Temp = random.randint(0, 256)
    message.sensor185Temp = random.randint(0, 256)
    message.sensor186Temp = random.randint(0, 256)
    message.sensor187Temp = random.randint(0, 256)
    message.sensor188Temp = random.randint(0, 256)
    message.sensor189Temp = random.randint(0, 256)
    message.sensor190Temp = random.randint(0, 256)
    message.sensor191Temp = random.randint(0, 256)
    message.sensor192Temp = random.randint(0, 256)
    message.sensor193Temp = random.randint(0, 256)
    message.sensor194Temp = random.randint(0, 256)
    message.sensor195Temp = random.randint(0, 256)
    message.sensor196Temp = random.randint(0, 256)
    message.sensor197Temp = random.randint(0, 256)
    message.sensor198Temp = random.randint(0, 256)
    message.sensor199Temp = random.randint(0, 256)
    message.sensor200Temp = random.randint(0, 256)
    message.sensor201Temp = random.randint(0, 256)
    message.sensor202Temp = random.randint(0, 256)
    message.sensor203Temp = random.randint(0, 256)
    message.sensor204Temp = random.randint(0, 256)
    message.sensor205Temp = random.randint(0, 256)
    message.sensor206Temp = random.randint(0, 256)
    message.sensor207Temp = random.randint(0, 256)
    message.sensor208Temp = random.randint(0, 256)
    message.sensor209Temp = random.randint(0, 256)
    message.sensor210Temp = random.randint(0, 256)
    message.sensor211Temp = random.randint(0, 256)
    message.sensor212Temp = random.randint(0, 256)
    message.sensor213Temp = random.randint(0, 256)
    message.sensor214Temp = random.randint(0, 256)
    message.sensor215Temp = random.randint(0, 256)
    message.sensor216Temp = random.randint(0, 256)
    message.sensor217Temp = random.randint(0, 256)
    message.sensor218Temp = random.randint(0, 256)
    message.sensor219Temp = random.randint(0, 256)
    message.sensor220Temp = random.randint(0, 256)
    message.sensor221Temp = random.randint(0, 256)
    message.sensor222Temp = random.randint(0, 256)
    message.sensor223Temp = random.randint(0, 256)
    message.sensor224Temp = random.randint(0, 256)
    message.sensor225Temp = random.randint(0, 256)
    message.sensor226Temp = random.randint(0, 256)
    message.sensor227Temp = random.randint(0, 256)
    message.sensor228Temp = random.randint(0, 256)
    message.sensor229Temp = random.randint(0, 256)
    message.sensor230Temp = random.randint(0, 256)
    message.sensor231Temp = random.randint(0, 256)
    message.sensor232Temp = random.randint(0, 256)
    message.sensor233Temp = random.randint(0, 256)
    message.sensor234Temp = random.randint(0, 256)
    message.sensor235Temp = random.randint(0, 256)
    message.sensor236Temp = random.randint(0, 256)
    message.sensor237Temp = random.randint(0, 256)
    message.sensor238Temp = random.randint(0, 256)
    message.sensor239Temp = random.randint(0, 256)
    message.sensor240Temp = random.randint(0, 256)
    message.sensor241Temp = random.randint(0, 256)
    message.sensor242Temp = random.randint(0, 256)
    message.sensor243Temp = random.randint(0, 256)
    message.sensor244Temp = random.randint(0, 256)
    message.sensor245Temp = random.randint(0, 256)
    message.sensor246Temp = random.randint(0, 256)
    message.sensor247Temp = random.randint(0, 256)
    message.sensor248Temp = random.randint(0, 256)
    message.sensor249Temp = random.randint(0, 256)
    message.sensor250Temp = random.randint(0, 256)
    message.sensor251Temp = random.randint(0, 256)
    message.sensor252Temp = random.randint(0, 256)
    message.sensor253Temp = random.randint(0, 256)
    message.sensor254Temp = random.randint(0, 256)
    message.sensor255Temp = random.randint(0, 256)
    return message


# print(message.ByteSize())
    # print(message.SerializeToString())
    # print(MessageToString(message))
    # return message

UDP_IP = '127.0.0.1'
UDP_PORT = 6000

s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, 0)
s.connect((UDP_IP, UDP_PORT))


# send 1 proto
temp = genProtoPacket()
print(temp)
temp = temp.SerializeToString()
teest = protoTest_pb2.telemetry()
teest.ParseFromString(temp)
print(teest)
s.send(temp)

# send 1 non proto
# temp = genNonProto()
# print(temp)
# temp = bytearray(temp)
# print(temp)
# print(len(temp))
# s.send(temp)
#